source/common/router/router.cc

```diff
@@ -462,6 +462,8 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   cluster_ = cluster->info();
 
   // Set up stat prefixes, etc.
+  route_entry_->finalizeRequestHeaders(headers, callbacks_->streamInfo(),
```

Comment:
Is this ok to move this here? Looking at the code below there places that use the Host header, which this function can modify.

Reply:
I think it should be fine. If virtualhost modifies host header (not sure if it is even allowed to) then it may be impacted but I doubt any one would that

Reply:
Maybe I misunderstood something. The `route_entry_->finalizeRequestHeaders(...)` changes the `headers` header map. Then there are a few places below where header values are read. For example line 476 or 525. Before this change these checks used the original values from the header map and now they would use new values. This maybe a breaking change.